rust/src/meas/randomized_response/mod.tex

```diff
@@ -0,0 +1,250 @@
+        return d_in * c;
+
+    def function(arg : bool) -> bool: |\label{line:fn}|
+        if (sample_bernoulli(prob, constant_time)): |\label{line:coinflip}|
+            return arg
+        else:
+            return !arg
+    
+    return Measurement(input_domain, output_domain, function, input_metric, similarity_metric, privacy_map)
+
+\end{lstlisting}
+%\hanwen{Just a small detail: in the proof \texttt{privacy\_map} is always referred to as \texttt{map}, should we change the name so that it's the same everywhere?}
+
+\begin{tcolorbox}
+\begin{warning}[Code is not constant-time]
+ \texttt{make\_randomized\_response\_bool} takes in a boolean \texttt{constant\_time} parameter that protects against timing attacks on the Bernoulli sampling procedure. However, the current implementation does not guard against other types of timing side-channels that can break differential privacy, e.g., non-constant time code execution due to branching.
+\end{warning}
+\end{tcolorbox}
+
+\section{Proof}
+The necessary definitions for the proof can be found at \href{https://github.com/opendp/whitepapers/blob/proof-defns/proof-defns/proof_defns.pdf}{``List of definitions used in the proofs"}. % may need to keep updating
+\begin{theorem} For every setting of the input parameters \texttt{prob, constant\_time} to \\ \texttt{randomized\_response} such that the given preconditions hold, \texttt{randomized\_response} raises an exception (at compile time or runtime) or returns a valid measurement with the following privacy guarantee:
+\begin{enumerate}  % we can discuss what a ``valid" measurement etc is
+    %\item \textup{(Appropriate randomization.)} The function that maps \texttt{input\_domain} to \texttt{output\_domain} achieves pseudorandomness. 
+    %\item \textup{(Appropriate output domain.)} For every element $v$ in \texttt{input\_domain}, \texttt{function}(v) is in \texttt{output\_domain}. 
+    %\item \textup{(Domain-metric compatibility.)} The domain \texttt{input\_domain} matches one of the possible domains listed in the definition of \texttt{input\_metric}, and the domain \texttt{output\_domain} matches one of the possible domains listed in the definition of \texttt{output\_metric}.
+    \item  \textup{(Domain-metric compatibility.)} The domain \texttt{input\_domain} matches one of the possible domains listed in the definition of \texttt{input\_metric}. 
+    
+    \item \textup{(Privacy guarantee.)} Let \texttt{d\_in} be the associated metric on \texttt{input\_domain} and has the associated type for \texttt{input\_metric}, and let $D$ be the similarity measure on probability distributions with the associated type for \texttt{similarity\_metric}. For every pair of elements $v, w$ in \texttt{input\_domain} and every \texttt{d\_in}, if $v, w$ are \texttt{d\_in}-close under \texttt{input\_metric}, then \texttt{function}(v), \texttt{function}(w) are \texttt{privacy\_map(d\_in)}-close with respect to $D$.
```

Comment:
I suggest the following rephrasing "Let \texttt{d\_in} have the associated type for \texttt{input\_metric}, and let $D$ have associated type for \texttt{similarity\_metric}."